app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoDumper.java

```diff
@@ -98,6 +101,52 @@ public AccessibilityNodeInfoDumper(@Nullable AccessibilityNodeInfo root,
         this.includedAttributes = includedAttributes;
     }
 
+    private int matchRootElementIndex() {
+        if (root == null) {
+            throw new RuntimeException(
+                    "Root node search query cannot be built if the root is unset"
+            );
+        }
+        for (int i = 0; i < uiElementsMapping.size(); ++i) {
+            if (Objects.equals(uiElementsMapping.valueAt(i).getNode(), root)) {
+                return uiElementsMapping.keyAt(i);
+            }
+        }
+        throw new RuntimeException("Cannot match the index of the root node");
+    }
+
+    @Nullable
+    private static Node matchRootElement(Node root, final int elementIndex) {
+        if (root instanceof Element
+                && ((Element) root).hasAttribute(UI_ELEMENT_INDEX)
+                && Integer.parseInt(((Element) root).getAttribute(UI_ELEMENT_INDEX)) == elementIndex
+        ) {
+            return root;
+        }
+        if (!root.hasChildNodes()) {
+            return null;
+        }
+        NodeList childNodes = root.getChildNodes();
+        for (int i = 0; i < childNodes.getLength(); ++i) {
+            Node result = matchRootElement(childNodes.item(i), elementIndex);
+            if (result != null) {
+                return result;
+            }
+        }
+        return null;
+    }
+
+    @NonNull
+    private Node fetchContext(InputStream xml) {
+        Document doc = loadDocument(xml);
+        return root == null || Settings.get(LimitXpathContextScope.class).getValue()
+                ? doc
+                : Objects.requireNonNull(
```

Comment:
okay.. so in this case current element can be considered as https://github.com/appium/appium-uiautomator2-server/pull/459#issuecomment-1024914217
btw, do we have test code related to this change?
Maybe in uia2-driver's e2e or this repo's test code may help our future

Reply:
this test would be a bit more complicated as we need to switch settings to check it. Currently e2e tests are disabled completely for all drivers since they need to be rewritten to comply with W3C/WDIO